internal/collector/cloudwatch.go

```diff
@@ -1,57 +1,93 @@
 	"github.com/openshift/cluster-logging-operator/internal/api/observability"
 	"github.com/openshift/cluster-logging-operator/internal/collector/common"
 	"github.com/openshift/cluster-logging-operator/internal/constants"
 	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/cloudwatch"
-	v1 "k8s.io/api/core/v1"
+	"github.com/openshift/cluster-logging-operator/internal/reconcile"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	CloudwatchCredentialsTemplate = template.Must(template.New("cw credentials").Parse(cloudwatchCredentialsTemplateStr))
+	//go:embed cloudwatch.credentials.tmpl
+	cloudwatchCredentialsTemplateStr string
 )
 
-// Add volumes and env vars if output type is cloudwatch and role is found in the secret
-func addWebIdentityForCloudwatch(collector *v1.Container, forwarderSpec obs.ClusterLogForwarderSpec, secrets observability.Secrets) {
-	if secrets == nil {
-		return
+type CloudwatchWebIdentity struct {
+	Name    string
+	RoleARN string
+	Token   string
+}
+
+// ReconcileAWSCredentialsConfigMap reconciles a configmap with credential profile(s) for Cloudwatch output(s).
+func ReconcileAWSCredentialsConfigMap(k8sClient client.Client, reader client.Reader, namespace, name string, outputs []obs.OutputSpec, secrets observability.Secrets, configMaps map[string]*corev1.ConfigMap, owner metav1.OwnerReference) (*corev1.ConfigMap, error) {
+	log.V(3).Info("generating AWS ConfigMap")
+	credString, err := GenerateCloudwatchCredentialProfiles(outputs, secrets)
+
+	if err != nil || credString == "" {
+		return nil, err
 	}
-	for _, o := range forwarderSpec.Outputs {
-		if o.Type == obs.OutputTypeCloudwatch && o.Cloudwatch.Authentication != nil && o.Cloudwatch.Authentication.Type == obs.CloudwatchAuthTypeIAMRole {
+	// The name is comprised of the forwarder name and aws-creds-configmap
+	cmName := name + "-" + constants.AWSCredentialsConfigMapName
+	configMap := runtime.NewConfigMap(
+		namespace,
+		cmName,
+		map[string]string{
+			constants.AWSCredentialsKey: credString,
+		})
+
+	utils.AddOwnerRefToObject(configMap, owner)
+	return configMap, reconcile.Configmap(k8sClient, reader, configMap)
+}
+
+// GenerateCloudwatchCredentialProfiles generates AWS CLI profiles for a credentials file from spec'd cloudwatch role ARNs and returns the formatted content as a string.
+func GenerateCloudwatchCredentialProfiles(outputs []obs.OutputSpec, secrets observability.Secrets) (string, error) {
+	// Gather all cloudwatch output's role_arns/tokens
+	webIds := GatherAWSWebIdentities(outputs, secrets)
 
+	// No CW outputs
+	if webIds == nil {
+		return "", nil
+	}
+
+	// Execute Go template to generate credential profile(s)
+	w := &strings.Builder{}
+	err := CloudwatchCredentialsTemplate.Execute(w, webIds)
```

Comment:
Will this return an array of  the template output?

Reply:
The template creates one final string that is a combination of all the web identities.
The string looks like the file `/internal/collector/cw_multiple_credentials`. This string is then written to the configmap